src/kernel/src/common/interrupt.c

```diff
@@ -77,13 +77,12 @@ void __attribute__((interrupt("UNDEF"))) undef_instruction_handler() {
 }
 
 long __attribute__((interrupt("SWI"))) software_interrupt_handler(void) {
-    int callNumber = 0, r0 = 0, r1 = 0, r2 = 0, r3 = 0;
-
-    asm volatile("MOV %0, r7" : "=r"(callNumber)::);
-    asm volatile("MOV %0, r0" : "=r"(r0)::);
-    asm volatile("MOV %0, r1" : "=r"(r1)::);
-    asm volatile("MOV %0, r2" : "=r"(r2)::);
-    asm volatile("MOV %0, r3" : "=r"(r3)::);
+    register int reg7 asm("r7");
```

Comment:
Didn't know you could do it this way. Looks useful

Reply:
We should still be careful when using it like that, because the registers are still not reserved for the specified variables. But at least this solves a bug that was present in the way it was previously done.

If you are interested in the bug, I added a comment with the disassembly here eb43463f31f155fb8d3c5ba4ab2a994a048450a5 (in the `src/kernel/src/common/interrupt.c` file). Basically, the compiler used r3 to store all the other variables while moving them in the inline assembly instructions, which resulted in the last variable being overwritten.